native_client/java/libdeepspeech/src/androidTest/java/org/mozilla/deepspeech/libdeepspeech/test/BasicTest.java

```diff
@@ -149,4 +151,33 @@ public void loadDeepSpeech_sttWithMetadata_withLM() {
         assertEquals("she had your dark suit in greasy wash water all year", decoded);
         m.freeModel();
     }
+
+    @Test
+    public void loadDeepSpeech_HotWord_withLM() {
+        DeepSpeechModel m = new DeepSpeechModel(modelFile);
+        m.enableExternalScorer(scorerFile);
+
+        for(int i = 0; i < word.length; i++) {
+            m.addHotWord(word[i], boost[i]);
+            String decoded = doSTT(m, false);
+            assertEquals("she had your dark suit in greasy wash water all year", decoded);
+            m.eraseHotWord(word[i]);
+        }
+
+        m.freeModel();
+    }
+
+    @Test
+    public void loadDeepSpeech_HotWord_noLM() {
+        DeepSpeechModel m = new DeepSpeechModel(modelFile);
+
+        for(int i = 0; i < word.length; i++) {
+            m.addHotWord(word[i], boost[i]);
+            String decoded = doSTT(m, false);
+            assertEquals("she had your dark suit in greasy wash water all year", decoded);
```

Comment:
make that a try/catch and assert his is `java.lang.RuntimeException: Error: External scorer is not enabled. (0x2004)`

Reply:
so we just take a single word and its boost value and check using try/catch if it is working or not.
if not, `Throw an RuntimeException`.

Reply:
nearly, except we want to catch the exception verify it this this error:
 - if it is, then the test SUCCEEDs
 - if it's a different exception, ERROR
 - if there's no exception, ERROR

Reply:
```
try {
            m.addHotWord(word[0], boost[0]);
            String decoded = doSTT(m, false);
            assertEquals("she had your dark suit in greasy wash water all year", decoded);
            
        }
        catch(RuntimeException e) {
            System.out.println("Error: External scorer is not enabled. (0x2004)");
        }
        finally {
            m.eraseHotWord(word[i]);
            m.freeModel();
        }
```
does it piece of code looks good?
@lissyx

Reply:
I'm no java wizard, but that looks good, except you should, in the `catch`:
 - make sure you have error 0x2004
 - if yes, nothing
 - if no, throw back the error

Reply:
try {
            m.addHotWord(word[0], boost[0]);
            String decoded = doSTT(m, false);
            assertEquals("she had your dark suit in greasy wash water all year", decoded);
        }
        catch(Exception e) {
            assertEquals(e, "Error: External scorer is not enabled. (0x2004)", e.getMessage());
        }
        finally {
            m.eraseHotWord(word[0]);
            m.freeModel();
        }

Reply:
@imrahul361 you don't need
```
            String decoded = doSTT(m, false);
            assertEquals("she had your dark suit in greasy wash water all year", decoded);
```

Just replace that with an `assert(false)`

Reply:
>         finally {
>             m.eraseHotWord(word[0]);
>             m.freeModel();
>         }

Please `assert(true)` after `freeModel` ?